great_expectations/expectations/expectation.py

```diff
@@ -794,38 +794,36 @@ def _diagnostic_unexpected_table_renderer(
             "partial_unexpected_counts"
         ):
             return None
-
         table_rows = []
 
-        if result_dict.get("partial_unexpected_counts"):
+        partial_unexpected_counts: Optional[List[dict]] = result_dict.get(
+            "partial_unexpected_counts"
+        )
+        if partial_unexpected_counts:
             # We will check to see whether we have *all* of the unexpected values
             # accounted for in our count, and include counts if we do. If we do not,
             # we will use this as simply a better (non-repeating) source of
             # "sampled" unexpected values
             total_count = 0
-            partial_unexpected_counts: Optional[List[dict]] = result_dict.get(
-                "partial_unexpected_counts"
-            )
-            if partial_unexpected_counts:
```

Comment:
Removed some redundant logic ( we were checking for `partial_unexpected_counts` twice)